rewrite-java-17/src/main/java/org/openrewrite/java/isolated/ReloadableJava17ParserVisitor.java

```diff
@@ -1290,7 +1290,7 @@ public J visitAnnotatedType(AnnotatedTypeTree node, Space fmt) {
             if (node.getUnderlyingType() instanceof JCFieldAccess) {
                 return new J.AnnotatedType(randomId(), fmt, Markers.EMPTY, leadingAnnotations, annotatedTypeTree(node.getUnderlyingType(), annotationPosTable));
             } else if (node.getUnderlyingType() instanceof JCArrayTypeTree) {
```

Comment:
Alternatively to my comment on line 1339 I suspect that this block here could be removed (didn't test this however).

Reply:
This is necessary, otherwise code will be parsed into whitespace.

In some of our parsers, we confirm that whitespace does not contain non-whitespace characters. I don't know if we ever added that to Java or if it was removed, but we don't check at this time.

I've found a handful of cases where source code was parsed into whitespace when it shouldn't have. So, it may be a good idea to add that to `Assertions#java`.